test/SerialIOTest.cpp

```diff
@@ -1528,6 +1596,51 @@ void optional_paths_110_test(const std::string & backend)
 
 
 #if openPMD_HAVE_HDF5
+TEST_CASE( "available_chunks_test_hdf5", "[serial][json]" )
+{
+    /*
+     * This test is HDF5 specific
+     * HDF5 does not store chunks explicitly,
+     * so the HDF5 backend will simply return the whole dataset as one chunk.
+     *
+     * Let's just write some random chunks and show that the HDF5 backend
+     * does not care.
+     */
+    constexpr unsigned height = 10;
+    std::string name = "../samples/available_chunks.h5";
+
+    std::vector< int > data{ 2, 4, 6, 8 };
+    {
+        Series write( name, Access::CREATE );
+        Iteration it0 = write.iterations[ 0 ];
+        auto E_x = it0.meshes[ "E" ][ "x" ];
+        E_x.resetDataset( { Datatype::INT, { height, 4 } } );
+        for( unsigned line = 2; line < 7; ++line )
+        {
+            E_x.storeChunk( data, { line, 0 }, { 1, 4 } );
+        }
+        for( unsigned line = 7; line < 9; ++line )
+        {
+            E_x.storeChunk( data, { line, 0 }, { 1, 2 } );
+        }
+        E_x.storeChunk( data, { 8, 3 }, {2, 1 } );
+        it0.close();
+    }
+
+    {
+        Series read( name, Access::READ_ONLY );
+        Iteration it0 = read.iterations[ 0 ];
+        auto E_x = it0.meshes[ "E" ][ "x" ];
+        ChunkTable table = E_x.availableChunks();
+        REQUIRE( table.size() == 1 );
+        /*
+         * Explicitly convert things to bool, so Catch doesn't get the splendid
+         * idea to print the Chunk struct.
```

Comment:
What does it need for that, an `ostream operator<<` implementation for Chunk? Could be generally useful for that type:
https://github.com/catchorg/Catch2/blob/devel/docs/tostring.md

Reply:
I don't find the failing builds any more, think they got lost in rebasing. It looked more like a linker bug in Catch that only came up in MSVC. g++ and clang had no issues with that.

Reply:
Ok, let's try to remove this work-around then and let us compare the vars directly.

Reply:
You mean, comparing the members of `Chunk`?

Reply:
I mean this should probably read:
```c++
        REQUIRE(
            table[ 0 ] == WrittenChunkInfo( { 0, 0 }, { height, 4 } ) );
```

Reply:
That's what I had originally, and I've re-run it to trigger the same error again: [MSVC](https://ci.appveyor.com/project/ax3l/openpmd-api/builds/36300565) doesn't like that.
```
SerialIOTest.obj : error LNK2019: unresolved external symbol "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const Catch::Detail::unprintableString" (?unprintableString@Detail@Catch@@3V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@B) referenced in function "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl Catch::Detail::convertUnstreamable<struct openPMD::WrittenChunkInfo>(struct openPMD::WrittenChunkInfo const &)" (??$convertUnstreamable@UWrittenChunkInfo@openPMD@@@Detail@Catch@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABUWrittenChunkInfo@openPMD@@@Z) [C:\projects\openpmd-api\build\SerialIOTests.vcxproj]
C:\projects\openpmd-api\build\bin\Release\SerialIOTests.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\openpmd-api\build\SerialIOTests.vcxproj]
```
For some reason, MSVC won't find [this symbol](https://github.com/openPMD/openPMD-api/blob/116f7ed29c254c7de8c28b95e3fc9f18a07b68ca/share/openPMD/thirdParty/catch2/include/catch2/catch.hpp#L12109), and I don't really know if it's really worth hunting down the reason for this error within a 15k line file of inlined code..?

Reply:
Discussed today: seems there is more than an `operator<<` to implement, so we stay with the prior work-around.

@franzpoeschel can you please update? :)